<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>XVideoUIClass</class>
 <widget class="QWidget" name="XVideoUIClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>799</width>
    <height>764</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>XVideoUI</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QPushButton:hover
 { 
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(120, 120, 120, 255), stop:0.495 rgba(39, 39, 39, 255), stop:0.505 rgba(39,39, 39, 255), stop:1 rgba(120, 120,120, 255));
 border: none;
border-radius:5px;
color: rgb(255, 255, 255);
font: 75 12pt &quot;黑体&quot;;
 }
QPushButton:!hover
 { 
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(39, 39, 39, 255), stop:0.495 rgba(120, 120, 120, 255), stop:0.505 rgba(120,120, 120, 255), stop:1 rgba(39, 39,39, 255));
 border: none;
border-radius:5px;
color: rgb(255, 255, 255);
font: 75 12pt &quot;黑体&quot;;
 }
#XVideoUIClass
{
	background-color: rgb(53, 53, 53);
}
#closeBtn
{
font: 75 11pt &quot;等线&quot;;
color: rgb(255, 255, 255);
background-color: rgb(53, 53, 53);
}

QLabel
{
	color:rgb(255, 255, 255);
	font: 75 12pt &quot;黑体&quot;;
}

#playBtn{background-color: rgba(255, 255, 255, 0);}
#playBtn:!hover{image: url(:/XVideoUI/Resources/PlayN.png);}
#playBtn:hover{image: url(:/XVideoUI/Resources/PlayH.png);}
#playBtn:pressed{image: url(:/XVideoUI/Resources/PlayP.png);}

#pauseBtn{background-color: rgba(255, 255, 255, 0);}
#pauseBtn:!hover{image: url(:/XVideoUI/Resources/PauseN.png);}
#pauseBtn:hover{image: url(:/XVideoUI/Resources/PauseH.png);}
#pauseBtn:pressed{image: url(:/XVideoUI/Resources/PauseP.png);}</string>
  </property>
  <widget class="QPushButton" name="closeBtn">
   <property name="geometry">
    <rect>
     <x>760</x>
     <y>0</y>
     <width>41</width>
     <height>31</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="text">
    <string>X</string>
   </property>
   <property name="flat">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="XVideoWidget" name="srcvideo1">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>50</y>
     <width>380</width>
     <height>280</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="openBtn">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>370</y>
     <width>61</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
  </widget>
  <widget class="QSlider" name="playSlider">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>360</y>
     <width>391</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>420</y>
     <width>111</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>亮度(1-100)</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>450</y>
     <width>151</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>对比度(1.0-3.0)</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="bright">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>420</y>
     <width>131</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="contrast">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>450</y>
     <width>131</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>1.000000000000000</double>
   </property>
   <property name="maximum">
    <double>3.000000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="setBtn">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>430</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>设置</string>
   </property>
  </widget>
  <widget class="XVideoWidget" name="desVideo">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>50</y>
     <width>380</width>
     <height>280</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="exportButton">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>370</y>
     <width>121</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>导出</string>
   </property>
  </widget>
  <widget class="QPushButton" name="playBtn">
   <property name="geometry">
    <rect>
     <x>100</x>
     <y>410</y>
     <width>61</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="pauseBtn">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>410</y>
     <width>91</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>480</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>图像旋转</string>
   </property>
  </widget>
  <widget class="QComboBox" name="rotate">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>480</y>
     <width>131</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>0</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>90</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>180</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>270</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>510</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>图像镜像</string>
   </property>
  </widget>
  <widget class="QComboBox" name="flip">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>510</y>
     <width>131</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>不处理</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>上下镜像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>左右镜像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>上下左右镜像</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>540</y>
     <width>111</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>图像尺寸W,H</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="width">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>540</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>99999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="height">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>540</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>99999</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>260</x>
     <y>570</y>
     <width>251</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>图像金字塔(高斯,拉不拉斯)</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="pyUp">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>570</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="pyDown">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>570</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>300</x>
     <y>600</y>
     <width>211</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>图像图像裁剪(x,y,w,h)</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="cx">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>600</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>99999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="cy">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>600</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>99999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="cw">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>600</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>99999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="ch">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>600</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>99999</number>
   </property>
  </widget>
  <widget class="QComboBox" name="color">
   <property name="geometry">
    <rect>
     <x>640</x>
     <y>480</y>
     <width>131</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>RGB图像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>灰度图</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string/>
    </property>
   </item>
  </widget>
  <widget class="QPushButton" name="markBtn">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>470</y>
     <width>111</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>水印(x,y,a)</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="my">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>470</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>99999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="mx">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>470</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>99999</number>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="ma">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>470</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
   <property name="value">
    <double>0.000000000000000</double>
   </property>
  </widget>
  <widget class="XVideoWidget" name="srcvideo2">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>510</y>
     <width>240</width>
     <height>180</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="blendBtn">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>700</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>融合</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="ba">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>710</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
   <property name="value">
    <double>0.500000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="mergeBtn">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>710</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>合并</string>
   </property>
  </widget>
  <widget class="QSlider" name="left">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>340</y>
     <width>391</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="value">
    <number>0</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QSlider" name="right">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>380</y>
     <width>391</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="value">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>XVideoWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>XVideoWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="XVideoUI.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>closeBtn</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>778</x>
     <y>18</y>
    </hint>
    <hint type="destinationlabel">
     <x>744</x>
     <y>647</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openBtn</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>451</x>
     <y>378</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderReleased()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>sliderRelease()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>161</x>
     <y>378</y>
    </hint>
    <hint type="destinationlabel">
     <x>159</x>
     <y>763</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderPressed()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>sliderPress()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>217</x>
     <y>373</y>
    </hint>
    <hint type="destinationlabel">
     <x>236</x>
     <y>758</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>setPos(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>294</x>
     <y>379</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>749</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setBtn</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>set()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>724</x>
     <y>446</y>
    </hint>
    <hint type="destinationlabel">
     <x>704</x>
     <y>742</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exportButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Export()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>605</x>
     <y>382</y>
    </hint>
    <hint type="destinationlabel">
     <x>684</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playBtn</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>play()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>423</y>
    </hint>
    <hint type="destinationlabel">
     <x>176</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pauseBtn</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>pause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>204</x>
     <y>431</y>
    </hint>
    <hint type="destinationlabel">
     <x>215</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>markBtn</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>mark()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>36</x>
     <y>478</y>
    </hint>
    <hint type="destinationlabel">
     <x>73</x>
     <y>-120</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blendBtn</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>blend()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>17</x>
     <y>720</y>
    </hint>
    <hint type="destinationlabel">
     <x>23</x>
     <y>871</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mergeBtn</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>merge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>716</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>872</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>left</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>left(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>352</x>
     <y>349</y>
    </hint>
    <hint type="destinationlabel">
     <x>343</x>
     <y>-80</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>right</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>right(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>56</x>
     <y>391</y>
    </hint>
    <hint type="destinationlabel">
     <x>88</x>
     <y>-84</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>open()</slot>
  <slot>sliderPress()</slot>
  <slot>sliderRelease()</slot>
  <slot>setPos(int)</slot>
  <slot>set()</slot>
  <slot>Export()</slot>
  <slot>play()</slot>
  <slot>pause()</slot>
  <slot>mark()</slot>
  <slot>blend()</slot>
  <slot>merge()</slot>
  <slot>left(int)</slot>
  <slot>right(int)</slot>
 </slots>
</ui>
